File: backend/app/schemas/article.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class ArticleResponse(BaseModel):
    id: uuid.UUID
    title: str | None
    authors: str | None
    journal: str | None
    year: int | None
    doi: str | None
    page_count: int | None
    status: str
    project_id: uuid.UUID | None
    uploaded_by: uuid.UUID
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class ArticleUpdate(BaseModel):
    title: str | None = None
    authors: str | None = None
    journal: str | None = None
    year: int | None = None
    doi: str | None = None
    project_id: uuid.UUID | None = None


class ArticleListResponse(BaseModel):
    articles: list[ArticleResponse]
    total: int


class PdfPageResponse(BaseModel):
    page_number: int
    width: float
    height: float
    text_content: str | None
    word_data: dict | None

    model_config = {"from_attributes": True}

File: frontend/src/types/article.ts
export interface Article {
  id: string;
  title: string | null;
  authors: string | null;
  journal: string | null;
  year: number | null;
  doi: string | null;
  page_count: number | null;
  status: string;
  project_id: string | null;
  uploaded_by: string;
  created_at: string;
  updated_at: string;
}

export interface PdfPage {
  page_number: number;
  width: number;
  height: number;
  text_content: string | null;
  word_data: WordData[] | null;
}

export interface WordData {
  text: string;
  x0: number;
  y0: number;
  x1: number;
  y1: number;
  block: number;
  line: number;
  word: number;
}
